src/index.ts

```diff
@@ -143,18 +147,60 @@ const initialize = (apiKey: string, target: Target, options?: Options): Result =
   }
 
   const authenticate = async (clientID: string, configuration: Options): Promise<string> => {
-    const response = await fetch(`${configuration.baseUrl}/client/auth`, {
+    const url = `${configuration.baseUrl}/client/auth`
+    const requestOptions: RequestInit = {
       method: 'POST',
       headers: { 'Content-Type': 'application/json', 'Harness-SDK-Info': SDK_INFO },
       body: JSON.stringify({
         apiKey: clientID,
         target: { ...target, identifier: String(target.identifier) }
       })
-    })
+    }
+
+    let timeoutId: any
+
+    if (window.AbortController) {
+      const abortController = new AbortController()
+      requestOptions.signal = abortController.signal
+
+      if (configurations.authRequestReadTimeout > 0) {
+        timeoutId = setTimeout(() => abortController.abort(), configuration.authRequestReadTimeout)
+      }
+
+      try {
+        const response = await fetch(url, requestOptions)
+
+        if (!response.ok) {
```

Comment:
We never checked the response was successful before.  We relied on flakey behaviour of parsing the JWT token which would result in an `invalid token` exception logged as the reason.  We now get to see the actual http error.